### PR TITLE
Io tools v3

### PR DESCRIPTION
**Pull request by mbelluzzo**
Changes from v2:
- Add missing return
- Add mission \n
- delete the timer
- wont use main macro for gpio, I dont want to run the mainloop

### CODE REVIEW

File: src/bin/sol-gpio/sol-gpio.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    fprintf(stdout,
+        "Usage:\n"
+        "   %s set [pin] [value]\n"
+        "   %s get [pin]]\n"
+        "   %s monitor [pin]\n",
+        program, program, program);
+}
+
+int
+main(int argc, char *argv[])
```
Comment (barbieri) [inline src/bin/sol-gpio/sol-gpio.c]: it doesn't matter the comment, the macro is to be used.

If you don't want to run the mainloop just call `sol_quit()` from `startup()`.

Comment (ibriano) [inline src/bin/sol-gpio/sol-gpio.c]: On Mon, 07 Mar 2016 13:12:26 -0800, Gustavo Sverzut Barbieri wrote:
> > +    sol_gpio_close(gpio);
> > +}
> > +
> > +static void
> > +usage(const char *program)
> > +{
> > +    fprintf(stdout,
> > +        "Usage:\n"
> > +        "   %s set [pin] [value]\n"
> > +        "   %s get [pin]]\n"
> > +        "   %s monitor [pin]\n",
> > +        program, program, program);
> > +}
> > +
> > +int
> > +main(int argc, char *argv[])
> 
> it doesn't matter the comment, the macro is to be used.
> 
> If you don't want to run the mainloop just call `sol_quit()` from `startup()`.

That doesn't work, if you call sol_quit() before the loop is running, it
complains and does nothing else.
Also, from what I remember, the main loop implementations will check if
they have to quit and maybe do nothing of what the tool does (no idea
what that is, didn't check the code). It'd need an idler to call quit
from it.

Comment (barbieri) [inline src/bin/sol-gpio/sol-gpio.c]: `startup()` is called from the mainloop, so it works

**Comment by mbelluzzo** [comment]
wow great, implement a startup function that just call sol_quit() just to use a macro ... great!

**Comment by barbieri** [comment]
You have the monitor in there, so instead of jumping over it, you quit. Not much different and yet provided us with uniformity and allows the macro to be changed without us having to remember to fix this example

**Comment by barbieri** [comment]
ping? Fix what I said and merge plz

**Comment by bdilly** [comment]
merged with proposed changes

**Pull request closed without merging**